src/dkim/mod.rs

```diff
@@ -66,10 +64,9 @@ impl Default for Canonicalization {
     }
 }
 
-#[derive(Debug, PartialEq, Eq, Clone)]
+#[derive(Clone)]
```

Comment:
Since the trait object can't easily implement `PartialEq` and `Debug`, I just removed these derives (since they didn't seem particularly important). We could probably add a manual `Debug` impl for `DomainKey` and maybe add a method to `VerifyingKey` that exposes the set of `Algorithm`s that it supports in addition to the flags, but not sure about the cost/benefit here.